agent/agent_test.go

```diff
@@ -128,3 +131,13 @@ func sameElements(a, b []string) bool {
 	}
 	return true
 }
+
+func TestTildeExpandRaceMetadata(t *testing.T) {
```

Comment:
Which is the assertion in this test?

Reply:
The test will panic with a race condition warning from golang in case of problem, if everything is ok, should finish normally. So no explicit assertion.

Reply:
The RC warning in go is something like this:

```
WARNING: DATA RACE
Read at 0x00c000436ba0 by goroutine 12:
  runtime.mapaccess1_faststr()
      /usr/local/go/src/runtime/map_faststr.go:12 +0x0
  go.undefinedlabs.com/scopeagent/agent.(*SpanRecorder).getPayload()
      /src/mn/projects/fullstory/go/src/go.undefinedlabs.com/scopeagent/agent/recorder.go:330 +0xa9f
  go.undefinedlabs.com/scopeagent/agent.(*SpanRecorder).sendSpans()
      /src/mn/projects/fullstory/go/src/go.undefinedlabs.com/scopeagent/agent/recorder.go:145 +0x10d
  go.undefinedlabs.com/scopeagent/agent.(*SpanRecorder).loop()
      /src/mn/projects/fullstory/go/src/go.undefinedlabs.com/scopeagent/agent/recorder.go:121 +0x29a
  go.undefinedlabs.com/scopeagent/agent.(*SpanRecorder).loop-fm()
      /src/mn/projects/fullstory/go/src/go.undefinedlabs.com/scopeagent/agent/recorder.go:80 +0x41
  gopkg.in/tomb%2ev2.(*Tomb).run()
      /src/mn/projects/fullstory/go/src/gopkg.in/tomb.v2/tomb.go:163 +0x38
Previous write at 0x00c000436ba0 by main goroutine:
  runtime.mapassign_faststr()
      /usr/local/go/src/runtime/map_faststr.go:190 +0x0
  go.undefinedlabs.com/scopeagent/agent.NewAgent()
      /src/mn/projects/fullstory/go/src/go.undefinedlabs.com/scopeagent/agent/agent.go:320 +0x1974
  go.undefinedlabs.com/scopeagent.Run()
      /src/mn/projects/fullstory/go/src/go.undefinedlabs.com/scopeagent/init.go:24 +0x10c
  go.undefinedlabs.com/scopeagent/autoinstrument.init.0.func1.1()
      /src/mn/projects/fullstory/go/src/go.undefinedlabs.com/scopeagent/autoinstrument/init.go:39 +0xc1
  fs/frontend_test.TestMain.func1()
      /src/mn/projects/fullstory/go/src/fs/frontend/setup_test.go:15 +0x62
  fs/frontend_test.TestMain()
      /src/mn/projects/fullstory/go/src/fs/frontend/setup_test.go:16 +0x38
  main.main()
      _testmain.go:108 +0x32e
Goroutine 12 (running) created at:
  gopkg.in/tomb%2ev2.(*Tomb).Go()
      /src/mn/projects/fullstory/go/src/gopkg.in/tomb.v2/tomb.go:159 +0x11c
  go.undefinedlabs.com/scopeagent/agent.NewSpanRecorder()
      /src/mn/projects/fullstory/go/src/go.undefinedlabs.com/scopeagent/agent/recorder.go:80 +0x578
  go.undefinedlabs.com/scopeagent/agent.NewAgent()
      /src/mn/projects/fullstory/go/src/go.undefinedlabs.com/scopeagent/agent/agent.go:285 +0x11ec
  go.undefinedlabs.com/scopeagent.Run()
      /src/mn/projects/fullstory/go/src/go.undefinedlabs.com/scopeagent/init.go:24 +0x10c
  go.undefinedlabs.com/scopeagent/autoinstrument.init.0.func1.1()
      /src/mn/projects/fullstory/go/src/go.undefinedlabs.com/scopeagent/autoinstrument/init.go:39 +0xc1
  fs/frontend_test.TestMain.func1()
      /src/mn/projects/fullstory/go/src/fs/frontend/setup_test.go:15 +0x62
  fs/frontend_test.TestMain()
      /src/mn/projects/fullstory/go/src/fs/frontend/setup_test.go:16 +0x38
  main.main()
      _testmain.go:108 +0x32e
==================
==================
WARNING: DATA RACE
Read at 0x00c000434878 by goroutine 12:
  github.com/vmihailenco/msgpack.encodeMapStringInterfaceValue()
      /src/mn/projects/fullstory/go/src/github.com/vmihailenco/msgpack/encode_map.go:71 +0x223
  github.com/vmihailenco/msgpack.(*Encoder).EncodeValue()
      /src/mn/projects/fullstory/go/src/github.com/vmihailenco/msgpack/encode.go:180 +0x99
  github.com/vmihailenco/msgpack.(*Encoder).Encode()
      /src/mn/projects/fullstory/go/src/github.com/vmihailenco/msgpack/encode.go:166 +0x122
  github.com/vmihailenco/msgpack.encodeMapStringInterfaceValue()
      /src/mn/projects/fullstory/go/src/github.com/vmihailenco/msgpack/encode_map.go:75 +0x2b5
  github.com/vmihailenco/msgpack.(*Encoder).EncodeValue()
      /src/mn/projects/fullstory/go/src/github.com/vmihailenco/msgpack/encode.go:180 +0x99
  github.com/vmihailenco/msgpack.(*Encoder).Encode()
      /src/mn/projects/fullstory/go/src/github.com/vmihailenco/msgpack/encode.go:166 +0x122
  github.com/vmihailenco/msgpack.Marshal()
      /src/mn/projects/fullstory/go/src/github.com/vmihailenco/msgpack/encode.go:55 +0x10d
  go.undefinedlabs.com/scopeagent/agent.encodePayload()
      /src/mn/projects/fullstory/go/src/go.undefinedlabs.com/scopeagent/agent/recorder.go:336 +0x5c
  go.undefinedlabs.com/scopeagent/agent.(*SpanRecorder).sendSpans()
      /src/mn/projects/fullstory/go/src/go.undefinedlabs.com/scopeagent/agent/recorder.go:147 +0x11b
  go.undefinedlabs.com/scopeagent/agent.(*SpanRecorder).loop()
      /src/mn/projects/fullstory/go/src/go.undefinedlabs.com/scopeagent/agent/recorder.go:121 +0x29a
  go.undefinedlabs.com/scopeagent/agent.(*SpanRecorder).loop-fm()
      /src/mn/projects/fullstory/go/src/go.undefinedlabs.com/scopeagent/agent/recorder.go:80 +0x41
  gopkg.in/tomb%2ev2.(*Tomb).run()
      /src/mn/projects/fullstory/go/src/gopkg.in/tomb.v2/tomb.go:163 +0x38
Previous write at 0x00c000434878 by main goroutine:
  go.undefinedlabs.com/scopeagent/agent.NewAgent()
      /src/mn/projects/fullstory/go/src/go.undefinedlabs.com/scopeagent/agent/agent.go:320 +0x198c
  go.undefinedlabs.com/scopeagent.Run()
      /src/mn/projects/fullstory/go/src/go.undefinedlabs.com/scopeagent/init.go:24 +0x10c
  go.undefinedlabs.com/scopeagent/autoinstrument.init.0.func1.1()
      /src/mn/projects/fullstory/go/src/go.undefinedlabs.com/scopeagent/autoinstrument/init.go:39 +0xc1
  fs/frontend_test.TestMain.func1()
      /src/mn/projects/fullstory/go/src/fs/frontend/setup_test.go:15 +0x62
  fs/frontend_test.TestMain()
      /src/mn/projects/fullstory/go/src/fs/frontend/setup_test.go:16 +0x38
  main.main()
      _testmain.go:108 +0x32e
Goroutine 12 (running) created at:
  gopkg.in/tomb%2ev2.(*Tomb).Go()
      /src/mn/projects/fullstory/go/src/gopkg.in/tomb.v2/tomb.go:159 +0x11c
  go.undefinedlabs.com/scopeagent/agent.NewSpanRecorder()
      /src/mn/projects/fullstory/go/src/go.undefinedlabs.com/scopeagent/agent/recorder.go:80 +0x578
  go.undefinedlabs.com/scopeagent/agent.NewAgent()
      /src/mn/projects/fullstory/go/src/go.undefinedlabs.com/scopeagent/agent/agent.go:285 +0x11ec
  go.undefinedlabs.com/scopeagent.Run()
      /src/mn/projects/fullstory/go/src/go.undefinedlabs.com/scopeagent/init.go:24 +0x10c
  go.undefinedlabs.com/scopeagent/autoinstrument.init.0.func1.1()
      /src/mn/projects/fullstory/go/src/go.undefinedlabs.com/scopeagent/autoinstrument/init.go:39 +0xc1
  fs/frontend_test.TestMain.func1()
      /src/mn/projects/fullstory/go/src/fs/frontend/setup_test.go:15 +0x62
  fs/frontend_test.TestMain()
      /src/mn/projects/fullstory/go/src/fs/frontend/setup_test.go:16 +0x38
  main.main()
      _testmain.go:108 +0x32e
==================
	14:15:15.925123 fs/frontend/memcache.go:35 DEBUG fetched "foobar" from source in 754ns
    logger.go:65: race detected during execution of test
```